Когда используется критерий Стьюдента, а когда Z –критерий?

Критерий Стьюдента используется:
- когда среднеквадратическое отклонение (сигма) генеральной совокупности неизвестна

Z –критерий используется:
- когда среднеквадратическое отклонение (сигма) генеральной совокупности известна 

Решать вручную: Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [15]:
x = 17
mu = 17.5
D = 4
s = D ** 0.5
n = 100
print((x - mu) / (s / n ** 0.5))

-2.5


In [1]:
from scipy import stats
a = 0.05
t_tabl = stats.norm.ppf(a)
print(t_tabl)

-1.6448536269514729


Т.к. -1.64 > -2.5, то утверждение, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм - не принимается

Решать вручную: Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. 

Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. 

Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

In [10]:
import numpy as np
arr = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
x = np.mean(arr)
mu = 200
s = np.std(arr, ddof = 1)
n = 10
print((x - mu) / (s / n ** 0.5))

-1.0651074037450896


In [11]:
from scipy import stats
a = (1 - 0.99) / 2
t_tabl = stats.t.ppf(a, n - 1)
print(t_tabl)

-3.2498355440153697


Т.к. -3.24 < -1.06, то нулевая гипотеза (утверждение продавца, что средний вес пачки печенья составляет 200 г) - принимается

Решать с помощью функции: Есть ли статистически значимые различия в росте дочерей?

Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [9]:
import numpy as np
from scipy import stats
x = np.array([172, 177, 158, 170, 178,175, 164, 160, 169])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])
stats.ttest_ind(x, y)

TtestResult(statistic=0.35581195161541973, pvalue=0.7266292679160495, df=16.0)

Так как p-value > alpha (0.72 > 0.05 (5%)), то нулевая гипотеза (статистически значимых различий в росте дочерей нет) - принимается